readthedocs/search/api.py

```diff
@@ -35,14 +37,37 @@ def get_link(self, obj):
     def get_highlight(self, obj):
         highlight = getattr(obj.meta, 'highlight', None)
         if highlight:
-            if hasattr(highlight, 'content'):
-                # Change results to turn newlines in highlight into periods
-                # https://github.com/rtfd/readthedocs.org/issues/5168
-                highlight.content = [result.replace('\n', '. ') for result in highlight.content]
-            ret = highlight.to_dict()
-            log.debug('API Search highlight: %s', pformat(ret))
+            ret = utils._remove_newlines_from_dict(highlight.to_dict())
+            log.debug('API Search highlight [Page title]: %s', pformat(ret))
```

Comment:
One of the good things of using `%s` formatting on `log.debug` is that the string is not rendered if DEBUG is not enabled. Here, since we are calling `pformat` over the object _before_ passing it to `log.debug` this is, in effect, formatting a _potential big object_ and then discarding it.

> There are other places where we are doing the same that we may want to consider as well.

Reply:
@humitos 
What should be the approach here?
Don't use pformat or there's an alternative to it?

Reply:
I don't know a way to have both things: pretty print and not consuming extra resources. Considering that we are facing a problem on re-index taking a lot of time, maybe it worth to remove `pformat` for now.

Maybe, the CPU saving is not noticeable, though. I suppose it worth a try.

Reply:
There are many places where we are using this.
Should we remove from all the places?

I grepped for `pformat`.
It is not being used in any of the indexing code.
But yeah -- each search result goes through it.

Reply:
I'd remove it, but I've never debugged this code and I don't know how `pformat` improves debugging here, so :man_shrugging: 

@ericholscher is probably the one with the :+1: or :-1: power. I'm :+1: on not using `pformat` when logging as a general rule.